frontend/exporter/src/types/new/full_def.rs

```
@@ -424,6 +425,100 @@ pub enum FullDefKind<Body> {
+    // Note: lifetime generics are allowed in vtable-safe methods.
+    let method_generics = tcx.generics_of(origin_trait_method_id);
+    
+    // Check if the method has its own type or const parameters (lifetimes are OK)
+    let has_own_type_or_const_params = method_generics.own_params.iter().any(|param| {
+        matches!(
+            param.kind,
+            ty::GenericParamDefKind::Type { .. } | ty::GenericParamDefKind::Const { .. }
+        )
+    });
+    
+    if has_own_type_or_const_params {
+        return None;
+    }
+    
+    let origin_trait_method_sig_binder = tcx.fn_sig(origin_trait_method_id);
+    
+    // Extract the trait reference from dyn_self
+    // dyn_self is of form `dyn Trait<Args...>`, we need to extract the trait args
+    match dyn_self.kind() {
+        ty::Dynamic(preds, _, _) => {
+            // Find the principal trait predicate
+            for pred in preds.iter() {
+                if let ty::ExistentialPredicate::Trait(trait_ref) = pred.skip_binder() {
+                    // Build full args: dyn_self + trait args
+                    // Note: trait_ref.args doesn't include Self (it's existential), so we prepend dyn_self
+                    let mut full_args = vec![ty::GenericArg::from(dyn_self)];
+                    full_args.extend(trait_ref.args.iter());
+                    
+                    let subst_args = tcx.mk_args(&full_args);
+                    
+                    // Instantiate the signature with the substitution args
+                    let origin_trait_method_sig = origin_trait_method_sig_binder.instantiate(tcx, subst_args);
+                    
+                    // Normalize the signature to resolve associated types
+                    let normalized_sig = normalize(tcx, s.typing_env(), origin_trait_method_sig);
+
+                    return Some(normalized_sig.sinto(s));
+                }
+            }
+            None
```

This should panic instead of returning `None`; `dyn_self_ty` guarantees that the first predicate is a trait predicate.